rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/Connection.kt

```diff
@@ -39,4 +40,7 @@ interface Connection : Cancellable {
 internal suspend fun Connection.receiveFrame(): Frame = receive().readFrame(pool)
 
 @OptIn(DangerousInternalIoApi::class, TransportApi::class)
-internal suspend fun Connection.sendFrame(frame: Frame): Unit = send(frame.toPacket(pool))
+internal suspend fun Connection.sendFrame(frame: Frame) {
+    val packet = frame.toPacket(pool)
+    packet.closeOnError { send(packet) }
```

Comment:
This is very unintuitive to me.  But understand it now.

Reply:
could we rename this to runWithCloseOnError?

Reply:
in one more coming PR signature will be changed to:
`fun <T : Closeable, R> T.closeOnError(block: (T) -> R): R`
Will rename it there